src/webapp/pages/settings/SettingsPage.tsx

```diff
@@ -81,9 +96,8 @@ export const SettingsPage: React.FC = () => {
             onSave: async () => {
                 loading.show(true, i18n.t("Deleting dangling documents"));
 
-                await usecases.instance.deleteDocuments(danglingDocuments.map(({ id }) => id));
-                const newDanglingList = await usecases.instance.listDanglingDocuments();
-                setDanglingDocuments(newDanglingList);
+                await usecases.document.delete(danglingDocuments.map(({ id }) => id));
+                setDanglingDocuments([]);
```

Comment:
If the delete fails, we will set the dangling documents as empty. Maybe we should append the `setDanglingDocuments` to the success branch of the promise

Reply:
@p3rcypj If the delete fails, setDanglingDocuments won't be reached, right?

Reply:
Certainly, when you said it again I just realized that if the promise failed it would be returned in the error branch because we are in an async. I was wrong. Thanks @tokland for pointing that out!